File: app/src/main/java/com/destructo/botox/HashUtil.java
package com.destructo.botox;

import android.os.Build;

import androidx.annotation.RequiresApi;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.HexFormat;

public class HashUtil {
    @RequiresApi(api = Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
    public static String hash(String data) {
        try {
            // SHA-256 algoritması için bir MessageDigest örneği oluşturulur
            MessageDigest digest = MessageDigest.getInstance("SHA-256");

            // Girdi verisini byte dizisine çevirir ve hash hesaplar
            byte[] hashBytes = digest.digest(data.getBytes(StandardCharsets.UTF_8));

            // Byte dizisini hex formata çevirir
            return  "a";//HexFormat.of().formatHex(hashBytes);

        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }
}

File: app/src/main/java/com/destructo/botox/HashUtil.kt
package com.destructo.botox

import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

object HashUtil {
     fun generateHash(resetCode: String): String {
        return try {
            val digest = MessageDigest.getInstance("SHA-256")
            val hash = digest.digest(resetCode.toByteArray())
            val hexString = StringBuilder()
            for (b in hash) {
                val hex = Integer.toHexString(0xff and b.toInt())
                if (hex.length == 1) hexString.append('0')
                hexString.append(hex)
            }
            hexString.toString()
        } catch (e: NoSuchAlgorithmException) {
            e.printStackTrace()
            ""
        }
    }
}